src/crates_cache.rs

```diff
@@ -137,6 +137,11 @@ impl CratesCache {
             request.call()
         };
 
+        if response.error() {
+            let err = response.synthetic_error().as_ref().unwrap();
+            return Err(io::Error::new(io::ErrorKind::Other, format!("{}", err)));
+        }
```

Comment:
Nit: The response could be an error without that error being synthetic. The real server could return an error.

Reply:
Good catch! Fixed, thanks!